src/app/features/tech-record/create-batch/components/batch-vehicle-results/batch-vehicle-results.component.ts

```diff
@@ -47,7 +47,7 @@ export class BatchVehicleResultsComponent implements OnDestroy {
   }
 
   get vehicleType(): VehicleTypes | undefined {
-    return this.techRecord ? this.technicalRecordService.getVehicleTypeWithSmallTrl(this.techRecord) : undefined;
+    return this.vehicleRecord ? (this.vehicleRecord.vehicleType as VehicleTypes) : undefined;
```

Comment:
do we not need the small trailer vehicle type on here ? 

Additionally, if `vehicleRecord` is only used to get the vehicle type, I would probably said to make a public property called `vehicleType` and removing `vehicleRecord` from the component.

Or you can do something like this, if vehicleType is only used in html:
```ts
get vehicleType$(): Observable<VehicleTypes | undefined> {
  this.batchRecordService.batchVehicles$.pipe(map(batchVehicle => batchvehicle?.vehicleType))
}
```
and then `async` pipe it in html

Reply:
changing this logic to save Vehicle type when selected the first time for batch creation and accessing from that on every page after